cmd/migrate/main_test.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"testing"
+
+	fakedownstreammachineclientset "github.com/kubermatic/machine-controller/pkg/client/clientset/versioned/fake"
+	machinev1alpha1downstream "github.com/kubermatic/machine-controller/pkg/machines/v1alpha1"
+
+	fakeclusterv1alpha1clientset "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/fake"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakekube "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestMigrateMachines(t *testing.T) {
+	testcases := []struct {
```

Comment:
would be nice to see / test the machine object after migration.

Reply:
The tests here are supposed to test the `migrateMachiens` func from `cmd/migrate` which itself uses `pkg/api/conversions` to do the actual conversions.

The conversion also has tests, you can check the before/after by looking into `pkg/api/conversions/testdata/downstreammachines`/`pkg/api/conversions/testdata/migrated` respectively. 

It is probably best to check this out in your local IDE where you have a tree view for the diff. Or I could split this into dedicated PRs for:

* Downgrade client-go
* `pkg/api/conversions`
* `cmd/migrate`

Would that make sense?

Reply:
yes, it does. thx.